### PR TITLE
Draft: ci(GitHub): add shellcheck 

### PR DESCRIPTION
**Pull request by Vincinator**
Adding shellcheck to GitHub actions

### CODE REVIEW

File: .github/workflows/shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: shellcheck
+on:
+  pull_request:
```
Comment (nkraetzschmar) [inline .github/workflows/shellcheck.yml:3]: maybe also on push? that way you can see something is wrong in your own branch before opening a PR

File: .github/workflows/shellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'     
+        required: true
+        default: 'warning'
+
+jobs:
+  shellcheck:
+    runs-on: gardenlinux-amd64
```
Comment (nkraetzschmar) [inline .github/workflows/shellcheck.yml:13]: we can probably just use the free github hosted runners (`runs-on: ubuntu-latest`) for this, shellcheck shouldn't need the additional performance of self hosted ones

File: .github/workflows/shellcheck.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        echo "Install shellcheck"
+        echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list \
```
Comment (nkraetzschmar) [inline .github/workflows/shellcheck.yml:18]: on the github hosted runners you can then probably just do update & install shellcheck from default sources

File: .github/workflows/shellcheck.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - run: shellcheck --version
+    - run: |
+        echo "Do shellcheck on all *.sh files"
+        status=0
+        set +e 
+        while IFS= read -r -d '' file
+        do
+            shellcheck --severity=warning "$file"  
+            status=$((status +$?))
+        done < <(find . -name '*.sh' -print0)
```
Comment (nkraetzschmar) [inline .github/workflows/shellcheck.yml:30]: won't this miss a lot of files? most of our scripts don't end in `.sh` so we'll probably need a more clever way to find all.... maybe find all files with executable bit set and then check the `#!` line 🤔

**Comment by nkraetzschmar** [comment]
or you could just cheat and use some pre-made github action like https://github.com/marketplace/actions/shellcheck 😅

File: .github/workflows/shellcheck.yml (hunk 2 above)
Comment (Vincinator) [reply .github/workflows/shellcheck.yml:13]: Unfortunately latest ubuntu image available for the GitHub runner is `20.04` and this has an older shellcheck.. but now that I am adding to sources.list anyway, I need to try again

File: .github/workflows/shellcheck.yml (hunk 3 above)
Comment (Vincinator) [reply .github/workflows/shellcheck.yml:18]: installs spellcheck 0.7.0

https://manpages.ubuntu.com/manpages/focal/man1/shellcheck.1.html

File: .github/workflows/shellcheck.yml (hunk 4 above)
Comment (Vincinator) [reply .github/workflows/shellcheck.yml:30]: good point

File: .github/workflows/shellcheck.yml (hunk 1 above)
Comment (Vincinator) [reply .github/workflows/shellcheck.yml:3]: > or you could just cheat and use some pre-made github action like https://github.com/marketplace/actions/shellcheck 😅

Does not work, only GitHub official or GardenLinux owned actions are allowed.. was my first try though :D

**Comment by Vincinator** [comment]
> or you could just cheat and use some pre-made github action like https://github.com/marketplace/actions/shellcheck 😅

Does not work, only GitHub official or GardenLinux owned actions are allowed.. was my first try though :D

**Comment by Vincinator** [comment]
Closing PR in favour for #725

**Pull request closed without merging**